### PR TITLE
Allow to configure Affinity or Tolerations for Backups and Restores

### PR DESCRIPTION
**Pull request by Smana**
### What problem does this PR solve? 

Fixes the issue https://github.com/pingcap/tidb-operator/issues/1535

### What is changed and how does it work?

Being able to set the affinity and tolerations in order to schedule backups or restores

### Does this PR introduce a user-facing change?:

```release-note
Allow users to configure affinity and tolerations for `Backup` and `Restore`
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ type TidbClusterSpec struct {
 	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
 
-	// Base tolerations of TiDB cluster Pods, components may add more tolreations upon this respectively
+	// Base tolerations of TiDB cluster Pods, components may add more tolerations upon this respectively
```
Comment (tennix) [inline pkg/apis/pingcap/v1alpha1/types.go]: Nice catch!

**Comment by tennix** [comment]
https://github.com/pingcap/tidb-operator/issues/1535 said both the helm chart and CRD based backup/restore need to set affinity and tolerations. Could you also add to the tidb-backup chart and tidb-cluster chart (scheduled backup).

**Comment by Smana** [comment]
Thx @tennix @cofyc 

>a reminder, hack/update-crd-groups.sh must run to update generated CRD yaml
done, thanks for the reminder

>#1535 said both the helm chart and CRD based backup/restore need to set affinity and tolerations. Could you also add to the tidb-backup chart and tidb-cluster chart (scheduled backup).
done

I've signed the CLA

**Comment by Smana** [comment]
@cofyc hello, please tell me if something is missing for this PR to be merged. Thanks

**Comment by Yisaer** [comment]
> @cofyc hello, please tell me if something is missing for this PR to be merged. Thanks

Hi @Smana, the sre-bot is doing the CI job to merge this request automatically, so there is nothing missing here.

**Comment by cofyc** [comment]
@Smana 
ah, our CI failed to trigger the job ... nothing is missing now

sre bot will merge this when CI is green.

**Comment by Smana** [comment]
Thanks @Yisaer

**Pull request merged**